src/Message/AbstractRequest.php

```diff
@@ -79,16 +79,31 @@ protected function getResponseBody(): array
      * @param string $requestMethod
      * @param string $urlPath
      * @param array  $data
+     * @param bool   $urlIsFullUrl  = false. True to have $urlPath be the absolute (full) url
      *
      * @return string
      */
-    private function getSecurityHash(string $requestMethod, string $urlPath, array $data): string
-    {
-        $string = $this->getBaseUrl().$urlPath.$requestMethod.$this->getContractProfileId().json_encode($data);
+    protected function getSecurityHash(
+        string $requestMethod,
+        string $urlPath,
+        array $data,
+        bool $urlIsFullUrl = false
+    ): string {
+        $contractProfileId = $this->getContractProfileId();
+        if (isset($data['ContractProfileId'])) {
+            $contractProfileId = $data['ContractProfileId'];
```

Comment:
How can this be different from the `ContractProfileId` that you are sending or do you want to make sure this is the contractProfileId that we are sending by checking it via the generated checksum?

Reply:
As discussed, the contractprofileId is not set as a request when we get a postback, so we need to inject it.